pyanaconda/exception.py

```diff
@@ -111,12 +111,16 @@ def _main_loop_handleException(self, dump_info):
                              "The exact error message is:\n\n%s.\n\n "
                              "The installer will now terminate.") % str(value)
             self.intf.messageWindow(_("Hardware error occurred"), hw_error_msg)
+            self._run_kickstart_scripts(dump_info)
```

Comment:
I think we should separate these two lines to a private method.

Reply:
I have created an issue for simplifying the code for exiting the installation. It is a little messy in general.